src/js/base/ripe.js

```diff
@@ -48,10 +49,15 @@ ripe.Ripe.prototype.init = function(brand, model, options) {
     };
     loadParts.call(this, function(result) {
         result = result || this.parts;
-        this.parts = result;
+        var parts = [];
+        for (var name in result) {
+            var part = result[name];
+            parts.push([name, part.material, part.color]);
+        }
+        this.setParts(parts);

```

Comment:
why are we changing to an array from the previous object structure ?

Reply:
setParts was not being used to set the parts here and it accepts an array. This is to keep the part events consistent